fc/mef.py

```diff
@@ -736,230 +506,223 @@ def get_transform_fxn(data_beads, peaks_mef, mef_channels,
-            else:
-                savefig = None
-            # Plot
+        if plot_dir is not None:
+            savefig = '{}/clustering_{}.png'.format(plot_dir, plot_filename)
+        else:
+            savefig = None
+
+        # If used one channel for clustering, make histogram
+        if len(clustering_channels) == 1:
+            plt.figure(figsize=(8,4))
+            fc.plot.hist1d(populations,
+                           channel=clustering_channels[0],
+                           div=4,
+                           alpha=0.75,
+                           savefig=savefig)
+
+        # If used two channels for clustering, make 2D scatter plot
+        elif len(clustering_channels) == 2:
             plt.figure(figsize=(6,4))
-            fc.plot.scatter2d(data_plot, 
-                              channels=cluster_channels,
+            fc.plot.scatter2d(populations,
+                              channels=clustering_channels,
                               savefig=savefig)
-            if plot_dir is not None:
-                plt.close()
-            
-        if len(cluster_channels) == 3:
-            if plot_dir is not None:
-                savefig = '{}/cluster_{}.png'.format(plot_dir, plot_filename)
-            else:
-                savefig = None
-            # Plot
+
+        # If used three channels or more for clustering, make 3D scatter plot
+        # with the first three.
+        elif len(clustering_channels) >= 3:
             plt.figure(figsize=(8,6))
-            fc.plot.scatter3d_and_projections(data_plot,
-                                              channels=cluster_channels,
+            fc.plot.scatter3d_and_projections(populations,
+                                              channels=clustering_channels[:3],
                                               savefig=savefig)
-            if plot_dir is not None:
-                plt.close()
 
-    # mef_channels and peaks_mef should be iterables.
-    if hasattr(mef_channels, '__iter__'):
-        mef_channel_all = list(mef_channels)
-        peaks_mef_all = np.array(peaks_mef).copy()
-    else:
-        mef_channel_all = [mef_channels]
-        peaks_mef_all = np.array([peaks_mef])
+        if plot_dir is not None:
+            plt.close()
 
     # Initialize lists to acumulate results
-    sc_all = []
-    if full:
-        peaks_ch_all = []
-        if find_peaks_method == 'smoothed_mode':
-            peaks_hists_all = []
-        sel_peaks_ch_all = []
-        sel_peaks_mef_all = []
-        sc_beads_all = []
-        sc_params_all =[]
+    std_crv_all = []
+    if full_output:
+        stats_values_all = []
+        selected_channel_all = []
+        selected_mef_all = []
+        beads_model_all = []
+        beads_params_all =[]
 
     # Iterate through each mef channel
-    for mef_channel, peaks_mef_channel in zip(mef_channel_all, peaks_mef_all):
+    for mef_channel, mef_values_channel in zip(mef_channel_all, mef_values_all):
```

Comment:
As for d627cebf3082816ff1e3a9d6887ede358556acbf, all `*_all` variables refer to things that will go in the output. I changed them to `*_out` in 2dfcc8e46baae36fd08a9e4bb212ab58e67cb5b6.